src/host/eigensolver.cpp

```diff
@@ -70,6 +70,40 @@ auto eigh(ContextInternal& ctx, T wl, ConstHostView<std::complex<T>, 2> s,
     }
   }
 
+  // Discard rows / columns with sum close to zero
+  for (std::size_t col = 0; col < s.shape(1); ++col) {
+    auto sum = s[{col, col}];
+    for (std::size_t row = col+1; row < s.shape(0); ++row) {
+      const auto val = s[{row, col}];
+      sum += val;
+    }
+    auto const row = col;
+    for (std::size_t col = 0; col < row; ++col) {
```

Comment:
The `col` name shadows the outer scope, which makes this a bit difficult to read.
But in general , we are now iterating three times through the matrix and it's a bit convoluted. Looking back the python parameter estimator, these checks look a bit different. 
Would it be enough to check if all elements are _close_ to zero instead?
If this would work, then something like this single loop should be enough:
```
  // flag working coloumns / rows
  std::size_t nVis = 0;
  for (std::size_t col = 0; col < s.shape(1); ++col) {
    for (std::size_t row = col; row < s.shape(0); ++row) {
      const auto val = s[{row, col}];
      if (std::norm(val) > std::numeric_limits<T>::epsilon()) {
        nonZeroIndexFlag[col] |= 1;
        nonZeroIndexFlag[row] |= 1;
        nVis += 1 + (row != col);
      }
    }
  }
```

Reply:
I tested your proposition on the problematic dataset for which the test on the sum over column/row of S was necessary (1st epoch of `LOFAR RX42_SB100-109.2ch10s.ms`) and it works.
I also tested on first epoch of MWA `1133149192-187-188_Sun_10s_cal.ms` where 34 columns out of 128 were flagged, and it performed the same. So for me that looks fine at this stage.
Will you suggest changes here and in the gpu counterpart or shall I include the changes myself?

Reply:
Great to hear it works for the datasets. It would be nice if you could change it here and in GPU version.

Reply:
Done in [20b3910](https://github.com/epfl-radio-astro/bipp/pull/28/commits/20b39105fb5c8f14477315b73b9dff73b3fc7ba3) and re-tested on OSKAR simulated data, LOFAR and MWA real data.